crates/solver/src/settlement_access_list.rs

```diff
@@ -1,35 +1,59 @@
-use ethcontract::{Address, H256};
+use anyhow::{anyhow, Result};
+use ethcontract::{dyns::DynTransport, transaction::TransactionBuilder, Address, H256};
 use reqwest::{
     header::{HeaderMap, HeaderValue},
     Client, Url,
 };
 use serde::{Deserialize, Serialize};
-use web3::types::Bytes;
+use shared::Web3;
+use web3::types::{AccessList, Bytes};
+
+#[async_trait::async_trait]
+pub trait AccessListEstimating: Send + Sync {
+    async fn estimate_access_list(
```

Comment:
The naming seems off.
Since there is actually no estimating involved `trait AccessListGenerating` or `trait AccessListing` would fit better?
Function names could then be `generate_access_list()` or `list_accesses()` (this one sounds a little weird, though).

Reply:
Hm, I was comparing this to the `eth_estimateGas` functionality which is also used for settlement simulation/submission. Also, with `estimation` in the name I was aiming to say that this trait does not necessarily return 100% precise and always the same result. The result could be different depending on the type of access list estimator and also the exact timing of the call.

Reply:
On suggestion would be `AccessListCreating` so that it is more 1:1 with the RPC method (`eth_createAccessList`).

That being said, I do think its fine as is.